src/execution/compiler/operator/hash_join_translator.cpp

```diff
@@ -14,7 +14,8 @@ HashJoinLeftTranslator::HashJoinLeftTranslator(const terrier::planner::HashJoinP
       hash_val_{codegen->NewIdentifier("hash_val")},
       build_struct_{codegen->NewIdentifier("BuildRow")},
       build_row_{codegen->NewIdentifier("build_row")},
-      join_ht_{codegen->NewIdentifier("join_ht")} {}
+      join_ht_{codegen->NewIdentifier("join_ht")},
+      left_semi_join_flag_{codegen->Context()->GetIdentifier("left_semi_join_flag")} {}
```

Comment:
This will be used for more than semi-joins. Suggest renaming to "mark" to support generic mark joins for semi, anti, left outer, etc.